pkgs/development/tools/misc/gf/default.nix

```diff
@@ -0,0 +1,66 @@
+{ lib, stdenv, makeWrapper
+
+  ,fetchgit, libX11, pkg-config, gdb
+
+  ,freetypeSupport ? false, freetype
+  ,extensionFolder ? null
+}:
+
+let
+  revision = "e0d6d2f59344f853a4a204d5313db6b6a5e5de7d";
+in
+
+stdenv.mkDerivation rec {
+  pname = "gf";
+  version = "2+${builtins.substring 0 7 revision}";
+
+  src = fetchgit {
+    url = "https://github.com/nakst/gf.git";
+    rev = "${revision}";
+    sha256 = "1b9iggz50l9aksblg1lj64daf2l5kjnkixdw8qjjblp63xz668r7";
+  };
+
+  nativeBuildInputs = [ makeWrapper pkg-config ];
+  buildInputs = [ libX11 gdb ]
+    ++ lib.optional freetypeSupport freetype;
+
+  inherit extensionFolder;
```

Comment:
What is this??

Reply:
The gf author provides extensions for purchase. The files of this extensions folder must be copied into the source directory. I wanted to add an option to add this extension folder. I'm still new to nix and probably there is a better option for doing this, but it worked for me. It is similar to how ST handles patches ([link](https://github.com/NixOS/nixpkgs/blob/62de360d75419386b69ea4af4a8ce104949a8242/pkgs/applications/terminal-emulators/st/default.nix#L26)).

Reply:
Well, I took some time reading the README.
I need to know a thing or two.

First, those third party extensions are loaded when the program is running, or they need to be present on build time?

Because, if they are to be loaded at runtime, then you do not need to include them at all - you just tell the running program to read them somehow.
This is not a case similar to ST, because ST patches explicitly modify the source code.

Else, the things are a bit more complicated. 

I suppose those extensions are handled by a proprietary license, and they can't be easily downloaded. There are some support for such proprietary things on Nixpkgs, however this is not recommended.

Reply:
The extensions are cpp files and need to be present during build time.

Reply:
Hum, that's interesting. It requires another approach.

I read the `build.sh` code, however I am not seeing how it detects and builds the extensions. Nothing like a `gcc *.cpp` on it.

Maybe the extension file also modifies the build script?

Until better judgement, remove all the extension stuff.

Reply:
The files are detected directly in the source. https://github.com/nakst/gf/blob/e0d6d2f59344f853a4a204d5313db6b6a5e5de7d/gf2.cpp#L1246-L1252

Reply:
What is wrong with the current approach? Are there any side effects?

Reply:
Now I understood it better. It uses preprocessor magic to include the extensions. It has the same effect as a patch file.
Now I have a better idea.